File: gestiondestock/src/main/java/fr/esic/gestiondestock/services/EntrepriseService.java
package fr.esic.gestiondestock.services;

import java.util.List;

import fr.esic.gestiondestock.dto.EntrepriseDto;

public interface EntrepriseService {
	
	EntrepriseDto save(EntrepriseDto dto);
	
	EntrepriseDto findById(Integer id);
	
	List<EntrepriseDto> findAll();
	
	void delete(Integer id);

}

File: frontend-gestion-de-stock/src/app/services/entreprise/entreprise.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { EntrepriseDto } from 'src/gs-api/src/models/entreprise-dto';
import { EntreprisesService } from 'src/gs-api/src/services/entreprises.service';

@Injectable({
  providedIn: 'root'
})
export class EntrepriseService {

  constructor(
    private entreprisesService: EntreprisesService
  ) { }

  sinscrire(entreprise: EntrepriseDto): Observable<EntrepriseDto> {
    return this.entreprisesService.save(entreprise);
  }

  findById(idEntreprise: number): Observable<EntrepriseDto> {
    return this.entreprisesService.findById(idEntreprise); 
  }
}
